url7895f5cfa7_edm92_defaultlogic_tgz-pJ8-be_fnord_DefaultLogic_AppJ8/src/be/fnord/DefaultLogic/DefaultLogicExample.java

```diff
@@ -0,0 +1,702 @@
+package be.fnord.DefaultLogic;
+
+import java.util.HashSet;
+
+import be.fnord.util.logic.DefaultReasoner;
+import be.fnord.util.logic.WFF;
+import be.fnord.util.logic.defaultLogic.DefaultRule;
+import be.fnord.util.logic.defaultLogic.RuleSet;
+import be.fnord.util.logic.defaultLogic.WorldSet;
+
+/**
+ * Default logic examples. Most of the examples here are from R. Reiter (1980).
+ * A logic for default reasoning. Artificial Intelligence, 13:81-132. Notes on
+ * output format - if you see the string eeee anywhere in the output take it as
+ * an EMPTY set or value. Default rules are written a:b ==> c where a is the
+ * prerequisite, b is justification and c is conclusion.
+ * ------------------------
+ * ------------------------------------------------------
+ * ------------------------------------------------
+ * <p/>
+ * I have made every attempt to test this reasoner; however, have not documented
+ * the decision proceedure yet so please take results with a grain of salt.
+ *
+ * @author Evan Morrison edm92@uowmail.edu.au http://www.fnord.be Apache
+ *         License, Version 2.0, Apache License Version 2.0, January 2004
+ *         http://www.apache.org/licenses/
+ */
+public class DefaultLogicExample {
+
+	public static void example1() {
+		WorldSet myWorld = new WorldSet();
+		myWorld.addFormula(a.e.EMPTY_EFFECT);
+
+		myWorld.addFormula("B"); // B is true
+		myWorld.addFormula("(A -> Z)"); // Extra rule to test closure
+		myWorld.addFormula("(C -> (D | A))"); // C implies either D or A or Both
+												// D and A
+		myWorld.addFormula("((A & C) -> ~E)"); // A and C implies not E
+
+		DefaultRule rule1 = new DefaultRule();
+		rule1.setPrerequisite(a.e.EMPTY_EFFECT);
+		rule1.setJustificatoin("A");
+		rule1.setConsequence("A");
+
+		DefaultRule rule2 = new DefaultRule();
+		rule2.setPrerequisite("B");
+		rule2.setJustificatoin("C");
+		rule2.setConsequence("C");
+
+		DefaultRule rule3 = new DefaultRule();
+		rule3.setPrerequisite("(D & A)");
+		rule3.setJustificatoin("E");
+		rule3.setConsequence("E");
+
+		DefaultRule rule4 = new DefaultRule();
+		rule4.setPrerequisite("C & E");
+		rule4.setJustificatoin("((~A) & (D | A))");
+		rule4.setConsequence("F");
+
+		RuleSet myRules = new RuleSet();
+		myRules.addRule(rule1);
+		myRules.addRule(rule2);
+		myRules.addRule(rule3);
+		myRules.addRule(rule4);
+
+		DefaultReasoner loader = new DefaultReasoner(myWorld, myRules);
+		HashSet<String> extensions = loader.getPossibleScenarios();
+
+		a.e.println("Given the world: \n\t" + myWorld.toString()
+				+ "\n And the rules \n\t" + myRules.toString());
+
+		a.e.println("Possible Extensions");
+		for (String c : extensions) {
+			a.e.println("\t Ext: Th(W U (" + c + "))");
+			// Added closure operator
+			a.e.incIndent();
+			WFF world_and_ext = new WFF("(( " + myWorld.getWorld() + " ) & ("
+					+ c + "))");
+			a.e.println("= " + world_and_ext.getClosure());
+			a.e.decIndent();
+
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 73:82, 120:129, 174:183, 222:231, 284:292, 333:342, 420:429, 465:474, 514:523, 564:573.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.